### PR TITLE
fix: updates required for latest hardhat and defender versions

### PR DESCRIPTION
**Pull request by ivelin**
See individual commit comments.

### CODE REVIEW

File: 05-upgrades-management/code/README.md [hunk 1]
```diff
@@ -10,14 +10,16 @@ Check out this folder to your local environment, and run:
 
 `npm install`
 
+`npm update`
```
Comment (mverzilli) [inline 05-upgrades-management/code/README.md:13]: This is too disruptive and env dependent. Let's update the package.json instead

**Review by mverzilli — CHANGES_REQUESTED** [review]
Thanks for the suggestions @ivelin! I wouldn't merge this telling people to `npm update`, since that really depends on the moment it is run. I prefer to pin the right versions on package.json so `npm install` will do the right thing.

**Comment by ivelin** [comment]
Makes sense. Thank you for reviewing the PR.

**Review by mverzilli — APPROVED** [review]
Looks good, thanks!

**Comment by ivelin** [comment]
Awesome, thanks!

**Pull request merged**